fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1336,6 +1338,101 @@ public void updateTableMeta(Database db, String tableName, Map<String, String> p
+            boolean binlogEnable = Boolean.parseBoolean(properties.get(
+                    PropertyAnalyzer.PROPERTIES_BINLOG_ENABLE));
+            if (binlogEnable != newBinlogConfig.getBinlogEnable()) {
+                newBinlogConfig.setBinlogEnable(binlogEnable);
+                hasChanged = true;
+            }
+        }
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_BINLOG_TTL)) {
+            Long binlogTtl = Long.parseLong(properties.get(
+                    PropertyAnalyzer.PROPERTIES_BINLOG_TTL));
+            if (binlogTtl != newBinlogConfig.getBinlogTtlSecond()) {
+                newBinlogConfig.setBinlogTtlSecond(binlogTtl);
+                hasChanged = true;
+            }
+        }
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE)) {
+            Long binlogMaxSize = Long.parseLong(properties.get(
+                    PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE));
+            if (binlogMaxSize != newBinlogConfig.getBinlogMaxSize()) {
+                newBinlogConfig.setBinlogMaxSize(binlogMaxSize);
+                hasChanged = true;
+            }
+        }
+        if (!hasChanged) {
+            if (context != null) {
+                context.setModifyFeMetaSuccess(true);
+            }
+            LOG.info("table " + olapTable.getName() +
+                    " binlog config is same as the previous version, so nothing need to do");
+            return;
+        }
+
+        if (olapTable.getCurBinlogConfig() != null) {
+            LOG.info("update binlog config of table " + olapTable.getName() +
+                    ", the current binlog is :" + olapTable.getCurBinlogConfig().toString());
+        } else {
+            LOG.info("update binlog config of table " + olapTable.getName() +
+                    ", it has no binlog config previously");
+        }
+
+        db.writeLock();
+        // check for concurrent modifications by version
+        if (olapTable.getBinlogVersion() != newBinlogConfig.getVersion()) {
+            // binlog config has been modified,
+            // no need to judge whether the binlog config is the same as previous one,
+            // just modify even if they are the same
+            Map<String, String> newProperties = olapTable.getCurBinlogConfig().toProperties();
+            newProperties.putAll(properties);
+            newBinlogConfig.buildFromProperties(newProperties);
+        }
+        newBinlogConfig.incVersion();
+        try {
+            GlobalStateMgr.getCurrentState().modifyBinlogMeta(db, olapTable, newBinlogConfig);
+            // context is null when alter table manually
+            if (context != null) {
+                context.setModifyFeMetaSuccess(true);
+            }
+        } finally {
+            db.writeUnlock();
+        }
+
+        // TODO optimize by asynchronous rpc
+        if (metaType != TTabletMetaType.DISABLE_BINLOG) {
```

Comment:
Why not update when disabling binlog?

Reply:
In order to return the result to mv quickly , the reporthanler is responsible for the final consistency